activerecord/lib/active_record/connection_adapters/connection_specification.rb

```diff
@@ -17,6 +17,37 @@ def to_hash
         @config.merge(name: @name)
       end
 
+      class ConnectionConfigurations < SimpleDelegator #:nodoc:
+        attr_accessor :root_level
```

Comment:
Since we're only using the instance variable there's no reason for the reader method. I'd do: `attr_writer`.